doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -185,129 +166,118 @@ Newly opened modules and sections inherit the current settings.
-   the current goal, theorems, axioms, etc) of the current context whose
-   statement contains :n:`@qualid`. This command is useful to remind the user
-   of the name of library lemmas.
+.. cmd:: Print Transparent Dependencies @smart_qualid
 
-   .. exn:: The reference @qualid was not found in the current environment.
+   Displays the assumptions and  transparent constants that :n:`@smart_qualid` depends on.
 
-      There is no constant in the environment named qualid.
+.. cmd:: Print All Dependencies @smart_qualid
 
-   .. cmdv:: Search @string
+   Displays all the assumptions and constants :n:`@smart_qualid` relies on.
 
-      If :n:`@string` is a valid identifier, this command
-      displays the name and type of all objects (theorems, axioms, etc) of
-      the current context whose name contains string. If string is a
-      notation’s string denoting some reference :n:`@qualid` (referred to by its
-      main symbol as in `"+"` or by its notation’s string as in `"_ + _"` or
-      `"_ 'U' _"`, see Section :ref:`notations`), the command works like ``Search`` :n:`@qualid`.
+.. todo: move Search* to the section on query commands once we agree on the wording
 
-   .. cmdv:: Search @string%@ident
+.. cmd:: Search {+ @search_item } {? {| inside | outside } {+ @qualid } }
 
-      The string string must be a notation or the main
-      symbol of a notation which is then interpreted in the scope bound to
-      the delimiting key :token:`ident` (see Section :ref:`LocalInterpretationRulesForNotations`).
+   .. insertprodn search_item search_item
 
-   .. cmdv:: Search @term_pattern
+   .. prodn::
+      search_item ::= {? - } @string {? @scope }
+      | {? - } @one_term
 
-      This searches for all statements or types of
-      definition that contains a subterm that matches the pattern
-      :token:`term_pattern` (holes of the pattern are either denoted by `_` or by
-      :n:`?@ident` when non linear patterns are expected).
+   Todo: :n:`@scope ::= % @ident` (fix in syntax extensions)
 
-   .. cmdv:: Search {+ {? -}@term_pattern_string}
+   Displays the name and type of objects matching :n:`@search_item`\s
+   in the selected goals (if a proof is open)
+   as well as accessible theorems, axioms, etc..
+   It's useful for finding the names of library lemmas.
 
-      where
-      :n:`@term_pattern_string` is a term_pattern, a string, or a string followed
-      by a scope delimiting key `%key`.  This generalization of ``Search`` searches
-      for all objects whose statement or type contains a subterm matching
-      :n:`@term_pattern` (or :n:`@qualid` if :n:`@string` is the notation for a reference
-      qualid) and whose name contains all string of the request that
-      correspond to valid identifiers. If a term_pattern or a string is
-      prefixed by `-`, the search excludes the objects that mention that
-      term_pattern or that string.
+   * :n:`@string` - If :n:`@string` is a substring of a valid identifier,
```

Comment:
I think there should be an introduction like "search_items may be of two kinds:"
In fact, I wonder whether the explanation on the kinds of search_items should not be split in *three* kinds, since string as subparts of identifiers and strings denoting notations are two different notions. And the explanation on the scope only applies to the latter.

Reply:
I think it's pretty clear there are 2 cases from the `@search_item` syntax.  Showing 3 cases makes the grammar ambiguous.  I added text to `% scope` saying it's only valid for notation strings.